pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +45,26 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var maxConcurrentReconciles int
 
 func init() {
 	flag.BoolVar(&runInTestMode, "test-mode", false, "disable ironic communication")
 	flag.BoolVar(&runInDemoMode, "demo-mode", false,
 		"use the demo provisioner to set host states")
+
+	if mcrEnv, ok := os.LookupEnv("OPERATOR_CONCURRENCY"); ok {
+		mcr, err := strconv.Atoi(mcrEnv)
+		if err == nil && mcr > 0 {
+			log.Info("OPERATOR_CONCURRENCY is set via an environment variable", mcr)
+			maxConcurrentReconciles = mcr
+		} else {
+			log.Error(err, "OPERATOR_CONCURRENCY value is invalid")
+			os.Exit(1)
```

Comment:
I think this is reasonable if somebody passes something that's not a number, but possibly a bit drastic if it's an empty string or e.g. -1? Both of those seem like common ways of resetting to default (perhaps -1 less so; it's more commonly used to specify 'unlimited', which doesn't really apply here). It can be surprisingly hard to unset an environment variable through several layers of configuration management tools, compared to just overriding.

Reply:
Now that you mention, that the env variable comes from many layers, perhaps it makes sense to rename the env variable to be something specific to BMO? Is exiting here a good idea?

Reply:
I think exiting is appropriate if somebody passes, you know, `OPERATOR_CONCURRENCY=penguins` or whatever.
Having a BMO-specific name does sound sensible, unless there is some sort of standard for this. @mhrivnak would know if there is.